### ISSUE TITLE
LspInfo inaccurate reporting attached buffer client details

### ISSUE THREAD

**Issue by lcrockett** [neovim/nvim-lspconfig#932]
# Description
Following earlier mentioned (and fixed) issue @ https://github.com/neovim/neovim/issues/14618 `:LspInfo` additionally doesn't consistently report details about the client(s) attached to a buffer. In the previously mentioned example (there's a video attached to the issue) using `:LspInfo` shows accurately reported client details for buffer #1. However, for buffer #2 + #3 it doesn't report any details about the buffer attached client (it actually is attached to the buffer). It does however report accurately the active clients all together.

# Images
Buffer #1
![image](https://user-images.githubusercontent.com/58056722/119248782-cbe06700-bb93-11eb-80d1-9de0662cb122.png)
Buffer #2
![image](https://user-images.githubusercontent.com/58056722/119248784-ce42c100-bb93-11eb-8a5c-5f2cd6fd0f54.png)
Buffer #3
![image](https://user-images.githubusercontent.com/58056722/119248786-d0a51b00-bb93-11eb-9ccd-a56f29089a8b.png)

**Comment by ahmedkhalf** [comment]
```diff
diff --git a/lua/lspconfig/lspinfo.lua b/lua/lspconfig/lspinfo.lua
index 9588ec6..532ae62 100644
--- a/lua/lspconfig/lspinfo.lua
+++ b/lua/lspconfig/lspinfo.lua
@@ -15,7 +15,7 @@ return function ()
   local buf_lines = {}
 
   local buf_client_names = {}
-  for _, client in ipairs(buf_clients) do
+  for _, client in pairs(buf_clients) do
     table.insert(buf_client_names, client.name)
   end
 

```

**Comment by ieh-12211-nfi** [comment]
@ahmedkhalf cheers for the pointers ! I found your change not be working for the issue mentioned above, however it looks like it couldn't hurt nonetheless. Together with a few other `ipairs` -> `pairs` changes i've managed to accurately report client information using `:LspInfo`. See below for code changes.

Would you like to file a PR or would you rather have me do it ?

```diff
diff --git a/lua/lspconfig/lspinfo.lua b/lua/lspconfig/lspinfo.lua
index 31d20a7..34320e4 100644
--- a/lua/lspconfig/lspinfo.lua
+++ b/lua/lspconfig/lspinfo.lua
@@ -15,7 +15,7 @@ return function ()
   local buf_lines = {}

   local buf_client_names = {}
-  for _, client in ipairs(buf_clients) do
+  for _, client in pairs(buf_clients) do
     table.insert(buf_client_names, client.name)
   end

@@ -29,7 +29,7 @@ return function ()

   local function trim_whitespace(cmd)
     local trimmed_cmd = {}
-    for _, str in ipairs(cmd) do
+    for _, str in pairs(cmd) do
       table.insert(trimmed_cmd, str:match'^%s*(.*)')
     end
     return trimmed_cmd
@@ -61,7 +61,7 @@ return function ()
     }
   end

-  for _, client in ipairs(buf_clients) do
+  for _, client in pairs(buf_clients) do
      local client_info = make_client_info(client)
      vim.list_extend(buf_lines, client_info)
   end
@@ -71,7 +71,7 @@ return function ()
     tostring(#clients).." active client(s): ",
   }
   vim.list_extend(buf_lines, active_section_header)
-  for _, client in ipairs(clients) do
+  for _, client in pairs(clients) do
      local client_info = make_client_info(client)
      vim.list_extend(buf_lines, client_info)
   end
@@ -95,7 +95,7 @@ return function ()
       cmd_is_executable = cmd
     end
     if config.filetypes then
-      for _, filetype_match in ipairs(config.filetypes) do
+      for _, filetype_match in pairs(config.filetypes) do
         if buffer_filetype == filetype_match then
           local matching_config_info = {
             indent.."",
```

**Comment by ahmedkhalf** [comment]
sure go ahead. but add me as co-author.
https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors

**Comment by mjlbach** [comment]
closed by #935
